database-plugins/src/main/java/io/cdap/plugin/db/batch/source/DBSource.java

```diff
@@ -158,7 +161,7 @@ private Connection getConnection() throws SQLException {
 
   @Override
   public void prepareRun(BatchSourceContext context) throws Exception {
-    sourceConfig.validate();
+    sourceConfig.validate(context.getFailureCollector());
```

Comment:
should call `collector.getOrThrowException` in prepareRun method to terminate further processing